src/core/core.js

```diff
@@ -442,7 +466,8 @@ Crafty.fn = Crafty.prototype = {
             }
         }
         delete this.__c[id];
-
+        // update map from component to (entityId -> entity)
```

Comment:
In the `Crafty()` selector we only need ids indeed. It could be changed to reflect this by having `{ componentName:  [id1, ..., idn] }`. However it might be useful later to retrieve entities of a component directly, without having to look up an entity from its id  (e.g. the suggested `getEntities` method).     
Which variant would you suggest here?